src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(state, obj.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const element of obj.keysToRemove) {
+          if (state.hasOwnProperty(element)) {
+            delete state[element];
+          }
+        }
+        break;
+
+      case 'clear':
+        for (const deleteEl in state) {
```

Comment:
```suggestion
        for (const key in state) {
```
the same here (p.3 checklist)